REVIEW_COPY.md

```diff
@@ -0,0 +1,312 @@
+            model
+
+        ( Stop, _ ) ->
+            { model | draggedVertex = Nothing }
+```
+
+And that's that! Again I skipped over the JavaScript that produces our messages and I promise we'll get to that in a moment. The full [polygon editor source code] is available if you're interested!
+
+## Last example: an outline editor
+
+Our final example is an outline editor. An outline is a tool for organizing our thoughts on a subject, but creating a list of concepts related to the thought, each of which have their own related thoughts, and so forth. The following image shows an example outline which can be re-arranged using drag & drop. We'll keep our scope small again by not bothering with creating and deleting nodes.
+
+![Rearranging outline nodes using drag & drop][outline.gif]
+
+We'll start by creating a model for our outline editor. It will need to keep track of two things: the outline itself and which node we're dragging.
+
+```elm
+type alias Model =
+    { outline : List OutlineNode
+    , draggedNode : Maybe DraggedNode
+    }
+
+type alias DraggedNode =
+    -- For simplicity sake we're going to use the node's contents as an id.
+    -- We get away with that here because we can ensure the nodes are unique:
+    -- the user will not be able to edit them in this example.
+    { node : String
+    , cursorOnScreen : Coords
+    , cursorOnDraggable : Coords
+    }
+
+type alias OutlineNode =
+    Tree String
+
+type Tree a
+    = Tree
+        { node : a
+        , children : List (Tree a)
+        }
+```
+
+Now we'll need to write behavior for the drag start, move, and end events.
+
+The drag starts when the user pressed down on a node. We can put an `onClick` handler on each node to detect when this happens. We'll skip the implementation in this post, but it's part of the full [outline editor source code] if you're interested!
+
+Then, as the user drags a node around we need to update that node's location in the outline. This part we're going to look at in detail.
+
+Lastly the drag stop event. We already changed the outline while the user was moving the cursor, and so all that's left to do here is change the model to its non-dragging state by setting `draggedNode` to `Nothing`.
+
+### Moving nodes in an outline
+
+Our hardest task is figuring out where to put our dragged node when the user moves the cursor. What might the user intent when dragging the cursor to a certain location? To move the dragged node in front of another node, behind it, or nested beneath it?
+
+Using dropzones we could draw invisible boxes in those positions that activate when the user moves over them, but the experience is unlikely to be great. Make the boxes too small and the user will not spend a lot of time over them, making the interface unresponsive for most of the time. Make the boxes too big and they start to overlap, causing the upper box instead of the closest one that will receive the dragged node.
```

Comment:
You wrote:
> Make the boxes too big and they start to overlap, causing the upper box instead of the closest one that will receive the dragged node.

I would suggest:
> Make the boxes too big and they start to overlap, causing the box in the uppermost layer to receive the dragged node even if it's not the closest.

This alleviates some of the confusion created by the multiple prepositional phrases at the end of your sentence.

Reply:
Great stuff, thank you!